pkg/storage/unified/search/bleve_mappings.go

```diff
@@ -137,6 +137,49 @@ func getBleveDocMappings(_ resource.SearchableDocumentFields) *mapping.DocumentM
 		IncludeInAll:       false,
 	})
 
+	referencesMapper := bleve.NewDocumentStaticMapping()
```

Comment:
🤔  rather than 4 sibling fields, what about:

* references.LibraryPanel = name

then it is a single field, and accurately gives us the search we need (it does not have "depends-on -- but that structure is not really useable for search anyway)